### PR TITLE
Move rustdoc header out of rustdoc container (Fixes #935)

### LINKED ISSUE #935: Header bar font is different on front page and rustdoc files
Regression introduced in #930.

rustdoc page
home page

@Nemo157 suggested this is because the writer puts the navigation nested under `.rustdoc` and the style sets the `.rustdoc` font differently. We probably just need an override resetting the `nav-bar` font style to what's used for the rest of the body outside rustdoc.

### PR DESCRIPTION
**Pull request by cynecx**
Closes #935.

### CODE REVIEW

**Comment by cynecx** [comment]
Uh, this is wrong.

**Comment by jyn514** [comment]
For the record:
```
$ git show HEAD~:templates/rustdoc/body.html | diff - templates/rustdoc/header.html 
238,270d237
< 
< <script type="text/javascript" src="/menu.js?{{ docsrs_version() | slugify }}"></script>
< <script type="text/javascript" src="/index.js?{{ docsrs_version() | slugify }}"></script>
< <script>
<       // Reset the scroll offset on browsers that don't support
<       // scroll-padding-top (Desktop & Mobile Safari):
<       const maybeFixupViewPortPosition = function() {
<         if (window.location.hash) {
<           const anchorElement = document.getElementById(window.location.hash.substr(1));
<           const navBarHeight = document.getElementsByClassName("nav-container-rustdoc")[0].offsetHeight;
<           if (anchorElement &&
<               anchorElement.getBoundingClientRect().top <= navBarHeight &&
<               Math.abs(anchorElement.getBoundingClientRect().top) >= 0) {
<             // It's just overlapped by the nav bar. This can't be a coincidence, scroll it into view:
<             requestAnimationFrame(function() {
<               scrollBy(0, -navBarHeight);
<             });
<           }
<         }
<       };
<       window.addEventListener("hashchange", maybeFixupViewPortPosition, false);
<       // Fix up the scroll position if this was a direct visit to the page
<       // (not back/forward navigation):
<       if (window.performance) {
<         const navEntries = window.performance.getEntriesByType('navigation');
<         const usedBack = navEntries.length > 0 && navEntries[0].type === 'back_forward' ||
<               (window.performance.navigation &&
<                window.performance.navigation.type == window.performance.navigation.TYPE_BACK_FORWARD);
<         if (!usedBack && window.location.hash) {
<           window.addEventListener("scroll", maybeFixupViewPortPosition, {"once": true});
<         }
<       }
< </script>
```
@cynecx is there a reason you added the extra `script`?

**Comment by cynecx** [comment]
@jyn514 Uh. I didn't add anything script related. I've just moved the html from one place to a new template file? Am I missing something?

**Comment by jyn514** [comment]
Oh I misread the diff - you kept the script in `body.html` and didn't add it to `header.html`. Yeah, that seems fine.

**Comment by cynecx** [comment]
@jyn514 Well I wasn't sure where changing the location of the script had any semantic difference, so I've just left it there :)

**Comment by jyn514** [comment]
🤦 I finally figured out the issue with my styles, I never ran a build with the new styles. Don't worry about getting a screenshot yourself, here's one:

image

**Comment by jyn514** [comment]
So this did fix the issue, nice :) looks good to me after nit.

File: src/utils/html.rs [hunk 1]
```diff
@@ -51,8 +52,11 @@ pub(crate) fn rewrite_lol(
         rustdoc_body_class.set_tag_name("div")?;
         // Prepend the tera content
         rustdoc_body_class.prepend(&tera_body, ContentType::Html);
-        // Now, make this a full <body> tag
+        // Wrap the tranformed body and rustdoc header into a <body> element
         rustdoc_body_class.before("<body>", ContentType::Html);
+        // Insert rustdoc's header
```
[the remark this answers is not in the document]
Comment (cynecx) [reply src/utils/html.rs]: Oh yeah my bad. I actually meant docsrs :)

**Comment by cynecx** [comment]
@jyn514 God damn it. You beat me. I was just running docker-compose xD.

**Comment by jyn514** [comment]
> @jyn514 God damn it. You beat me. I was just running docker-compose xD.

Well, I did already have the full build cached :P

**Pull request merged**